idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/utils/IdeUtils.kt

```diff
@@ -185,6 +189,41 @@ fun PsiElement.ctx(): CompilerContext? =
 fun Project.ctx(): CompilerContext? =
   getService(CompilerContext::class.java)
 
+fun <P : PsiElement> AnnotationBuilder.registerLocalFix(
+  fix: LocalQuickFix,
+  psi: P,
+  message: String,
+  highlightType: ProblemHighlightType = ProblemHighlightType.INFORMATION,
+  onThefly: Boolean = true,
+  afterBorderLine: Boolean = true
+): AnnotationBuilder.FixBuilder =
+  newLocalQuickFix(fix, psi.inspectionManager()
+    .createProblemDescriptor(psi, message, arrayOf(fix), highlightType, onThefly, afterBorderLine))
+
+fun <P : PsiElement> descriptor(
+  element: P,
+  message: String,
+  fix: LocalQuickFix? = null,
+  highlightType: ProblemHighlightType = ProblemHighlightType.INFORMATION,
+  onThefly: Boolean = true
+): ProblemDescriptor =
+  element.inspectionManager().createProblemDescriptor(element, message, fix, highlightType, onThefly)
```

Comment:
the following function is from the branch, I mentioned:
```suggestion
fun <P : PsiElement> Annotation.descriptor(element: P, fix: LocalQuickFix? = null, onThefly: Boolean = true): ProblemDescriptor =
  element.inspectionManager().createProblemDescriptor(element, message, fix, highlightType, onThefly)
```

Reply:
where are you getting the `message` and `highlightType` from?

Reply:
check the receiver type

Reply:
This wasn't needed in the end.. by doing `AnnotationBuilder.registerLocalFix` the `ProblemDescriptor` is created already. Actually the API for creating `Annotation`'s has been deprecated.. so I'm not sure if that helper is needed.

Reply:
It depends on what API one is using. In other API's (who aren't deprecated) users need to pass Problemdescriptors to functions.
But no need to add it in this PR 😄, when it has no use.